### PR TITLE
Use protobuf embeded in SL-CORE and other small fixes

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/EclipseInputFile.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public Charset getCharset() {
+    try {
+      return Charset.forName(file.getCharset());
+    } catch (CoreException e) {
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/EclipseInputFile.java]: MAJOR Either log or rethrow this exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AnalyzeProjectJob.java#L208](https://github.com/SonarSource/sonarlint-eclipse/blob/7dbe0238dadbd6d00f7ed3a048a6882aed5a6908/org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java#L208): Make "buildInputFiles" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: pom.xml [hunk 2]
```diff
@@ -141,11 +141,11 @@
       <artifactId>jsr305</artifactId>
       <version>3.0.1</version>
     </dependency>
-    <dependency>
+    <!--dependency>
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
       <version>${protobuf.version}</version>
-    </dependency>
+    </dependency-->
```
Comment (janos-ss) [inline pom.xml]: And the `com.diffplug.osgi.extension.sun.misc`

**Comment by shuzuo** [comment]
How to deal with this problem.
tks!

Unable to parse source file : G:\ykysvn\search\ictrade-search-service\src\main\java\com\ictrade\elasticsearch\ElasticsearchClient.java
Parse error at line 271 column 20:

261:  public SearchResponse search(String index,String type,String json,
262:    String[] aggs,List<SortBuilder> orders,int from,int size) throws UnknownHostException{
263:   Client client = null;
264:   if(size>MAX_PAGE_SIZE)size = MAX_PAGE_SIZE;
265:   try {
266:    client = getConnection();//鑾峰彇杩炴帴
267:    SearchRequestBuilder searchRequest = getSearchRequest(client, index, type, json, aggs, orders, from, size);
268:    logger.info("寮�鎼滅储");
269:    logger.debug("鎼滅储鍐呭锛�+searchRequest.toString());
270:    SearchResponse response = searchRequest.execute().actionGet();//鎵ц鎼滅储
271:       logger.info("鎼滅储鍒帮細"+response.getHits().getTotalHits());
                        ^
272:       logger.debug("妫�储缁撴灉锛�+response);

**Comment by dbmeneses** [comment]
@shuzuo please report the problem in the SonarLint forum: https://groups.google.com/forum/#!forum/sonarlint.

**Pull request closed without merging**